examples/caching_and_replay/cacheablemodel.py

```diff
@@ -0,0 +1,28 @@
+from model import Schelling
+from mesa_replay import CacheableModel, CacheState
+
+
+class CacheableSchelling(CacheableModel):
+    """A wrapper around the original Schelling model to make the simulation cacheable and replay-able.
+    Uses CacheableModel from the Mesa-Replay library, which is a decorator that can be put around any regular mesa model
```

Comment:
I don't see any decorator used in this example.

Reply:
Sorry, I don’t really understand. What do you mean you don’t see any decorator in this example? What would you expect to see instead?
From an sw architectural point of view, CacheableModel uses the decorator pattern and it is being inherited from here. Do we have a different understanding of a decorator? Is there some Python specific usage of the when decorator?

Reply:
>  Is there some Python specific definition of the term decorator?

It's a [Python decorator](https://en.wikipedia.org/wiki/Python_syntax_and_semantics#Decorators), and is the first definition that comes to people's mind, at least for Python users.

Reply:
Ah, that’s good to know. Maybe I can use those Python decorators to improve Mesa-Replay.

Reply:
I'd say "wrapper" is sufficient. Saying "decorator" to mean what you meant may be fine in other languages, but it's misleading in Python.

Reply:
Yeah, I agree. Changing decorator to wrapper in the description should be sufficient and fine. I was just thinking if it makes sense to actually use the Python decorator functionality to make ModelCacheable a little more accessible. This I will test out today in the evening. 

Just to name one example: Right now, ModelCacheable explicitly passes the attribute calls it gets to the model, to make it behave like the model from the user’s perspective. When using Python decorators this may no longer be needed, because then the object would still be the model itself, now just with added functionality. So Python decorators might make it possible to simplify the code a little.

Reply:
Python decorators are meant to modify a function/class on a very specific/limited scope. If `ModelCacheable` were to be made a decorator, it would add too much functionality. So far we have `run_model_until_condition_met`, but who knows what features are added later. This is debatable, though.

I think `CacheableModel` is similar to Mesa's `batch_run`, in that it operates on a model object.